### ISSUE TITLE
Permission to add dhewm3 to Steam as free DLC for RetroArch

### ISSUE THREAD

**Issue by twinaphex** [dhewm/dhewm3#270]
Hi there @DanielGibson ,

we would like to seek your permission to include our libretro core version of dhewm3 as a core as free DLC for RetroArch on Steam.

The user would have to manually install this, it would be freely made available.

If there's anything you are unsure of or are hesitant about, don't hesitate to bring it to our attention. We will always go with whatever you decide.

In case you are not aware of what this is all about, you can find our repo here -

https://github.com/libretro/dhewm3-libretro

And you can find more details on our project here -

www.libretro.com
www.retroarch.com

Our Github repository (we are a GPLv3 project) can be found here -

https://github.com/libretro/RetroArch

**Comment by DanielGibson** [comment]
Hi,
I don't think I could prevent you from doing it even if I wanted, as the code is under GPL - and in fact I don't mind you using it.

However, please don't use the name "dhewm3", so people know it's separate projects - I fear that otherwise I'll get bugreports for your version (that might not be reproducible in original dhewm3) by confused users. (Of course I don't mind collaborating on bugs that indeed affect both versions)

Cheers,
Daniel

**Comment by twinaphex** [comment]
If our sincere intent here to uplift open source projects by letting them have a page on Steam where they can get more recognition for their work and where people can learn more about their project is not appreciated, we can always just not publish the core as DLC.

I fear that our intent here is getting misunderstood based on the interactions with somebody from ioquake3. We are trying to do the right thing by contacting everybody involved in the code chain and getting their blessing. So far, we have 42 approved cores, including Tyr Quake from Tyrann. 

If, for whatever reason, you are not proud of the work we've done with the libretro fork and you kinda begrudgingly want to allow it just by virtue of the license allowing for it, if you want, we can just not publish it. We have no intent on starting drama here with anybody from upstream. We come here in peace, hoping to meet fellow open source developers to collaborate with. It's not our intent to be antagonistic.

> However, please don't use the name "dhewm3", so people know it's separate projects - I fear that otherwise I'll get bugreports for your version (that might not be reproducible in original dhewm3) by confused users. 

The changes were probably necessary to ensure it would work within the confines of a libretro core. If there's any interest in working on this so the patches can be pushed upstream, let us know. Whatever we do, we do out of the interests of getting the core to work as well as it can.

**Comment by DanielGibson** [comment]
As I said, I don't mind you using it, I'd just prefer if you used another name so people know they should report bugs to you first - and if you find out they're also present in latest dhewm3, we can try to figure them out together

**Comment by twinaphex** [comment]
Well, your response on Twitter seemed to indicate you didn't kinda like the idea, so I was just wondering if we were doing something against your will and something you don't feel comfortable with.

And of course, if the option is there, we'd always prefer working with upstream. The only times when something has to become a hard fork is when there are certain codebase changes that are necessary in order to get it to work on a specific platform or if the patch set to get it to work on libretro is so big that upstream doesn't want to take responsibility for it. In any case, we're just trying to be cooperative here to the best of our ability, and just seeing what willingness there is on the other end to come to some kind of symbiotic solution.  There's no willingness here to run roughshot over anybody.

**Comment by DanielGibson** [comment]
> Well, your response on Twitter seemed to indicate you didn't kinda like the idea

Based on the experience Jack Slater had with someone else from your team (who apparently said you'd use the code anyway which is a weird move esp. after asking) I wasn't very thrilled by the idea.
However, you seem to be more friendly and cooperative :-)

Regarding upstreaming: You seem to have replaced SDL with libretro which is a quite invasive change I have no interest in getting upstream, so I guess you'll have to keep maintaining a fork (btw, I wonder why you didn't just fork this repo and base you changes on it? If the dhewm3 commit history was still present, merging changes would be easier, probably).
If you have any other changes that make sense in dhewm3, I'd be happy to take a closer look at them and maybe merge them though!

I saw support for 360Hz/fps somewhere - did you do any changes to support this properly? Because right now dhewm3 only supports 60Hz and the experimental `com_fixedTic -1` hack turned out to be buggy and I didn't have time to come up with a proper solution for higher refreshrates yet.

**Comment by twinaphex** [comment]
> Based on the experience Jack Slater had with someone else from your team (who apparently said you'd use the code anyway which is a weird move esp. after asking)  I wasn't very thrilled by the idea. However, you seem to be more friendly and cooperative :-)

You're talking to the very same guy, and I've since talked to several people who told me about their own personal experiences with him. You don't have to trust me, but based on the weird conversation I had with him and the tangents he went to, I would take whatever he says personally with a pinch of salt. We also have no interests in engaging in any childish open source drama, and whatever drama happened in that scene is none of our interests, we just politely went to every author we could think of to see if they'd have an issue with us distributing vitaQuake 3 as a free DLC. Even though this is a complete hard fork of ioquake 3 at this point and the GPL allows it, we are not going to be distributing it period, and we told that to him before he started to make drama on Twitter. So honestly, we blocked, moved on, made our decision to not release it, but all of this could have been accomplished without this childish posturing and chest beating. We were not at fault there and it's very much this particular attitude that is holding open source development back from getting the recognition that it deserves in the wider world. 

> Regarding upstreaming: You seem to have replaced SDL with libretro which is a quite invasive change I have no interest in getting upstream, so I guess you'll have to keep maintaining a fork (btw, I wonder why you didn't just fork this repo and base you changes on it? If the dhewm3 commit history was still present, merging changes would be easier, probably).
If you have any other changes that make sense in dhewm3, I'd be happy to take a closer look at them and maybe merge them though!

Rinnegatamante did 90 to 95% of the port. We were just focused on getting it working.

Regarding SDL - well that is a particular pet peeve that we are just not always going to agree on. I understand why some devs swear by it, but to us it just seems like a big kitchensink API that from our libretro API perspective is best getting rid of altogether. We follow a quite different app design philosophy there. Anyway, we don't want to cast aspersions on SDL or get into any kind of competition with them - we just have our own reasons for reimplementing things, primarily because libretro operates on a frame-based runloop, so often times a codebase needs some alterations so it can run optimally.

> I saw support for 360Hz/fps mentioned somewhere 

Framerates above 60fps just use com_fixedTic - 1. It might be more buggy but it's also the only way to play it at higher framerates so it's better than nothing I wager.
